File: frontend/template/web/template-1-web/src/dto/user.dto.ts
import type { AppUserRole } from './enums';

export interface AppUserDTO {
  id: number;
  name: string;
  email: string;
  phone: string | null;
  displayName: string;
  roles: AppUserRole[];
  createdAt: string;
  updatedAt: string | null;
}

export interface AppUserRequestDTO {
  name: string;
  email: string;
  phone?: string;
  displayName: string;
  password: string;
  roles: AppUserRole[];
}

export interface AppUserUpdateDTO {
  name: string;
  email: string;
  phone?: string;
  displayName: string;
  password: string;
  roles: AppUserRole[];
}

export interface UpdateRolesRequestDTO {
  roles: AppUserRole[];
}

File: backend/office/expenditure-service/src/main/java/project/office_expenditure_service/model/dto/object/UserDTO.java
package project.office_expenditure_service.model.dto.object;

import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Null;
import lombok.Builder;
import lombok.Data;
import project.office_expenditure_service.model.entity.enums.UserRole;

import java.time.LocalDateTime;
import java.util.Set;

@Data
@Builder
public class UserDTO {
    @Null
    private Long id;
    private String uid;
    private String name;
    private String email;
    private String displayName;
    @NotNull
    private Set<UserRole> roles;
    @Null
    private LocalDateTime createdAt;
    @Null
    private LocalDateTime updatedAt;
}
